src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -17,211 +17,233 @@ namespace Castle.MicroKernel
 	using System;
 	using System.Collections;
 	using System.Collections.Generic;
+	using System.Linq;
 
 	using Castle.Core;
-	using Castle.MicroKernel.Context;
 
 	/// <summary>
-	///   Represents collection of arguments used when resolving a component.
+	/// Represents collection of arguments used when resolving a component.
 	/// </summary>
 	public class Arguments : IDictionary
 	{
-		protected IDictionary arguments;
+		private bool isReadOnly;
+		private IDictionary dictionary;
 
-		public Arguments(object namedArgumentsAsAnonymousType, params IArgumentsComparer[] customComparers)
-			: this(new ReflectionBasedDictionaryAdapter(namedArgumentsAsAnonymousType), customComparers)
-		{
-		}
-
-		public Arguments(IDictionary values, params IArgumentsComparer[] customComparers)
-			: this(customComparers)
+		public Arguments(IDictionary values)
+			: this()
 		{
 			foreach (DictionaryEntry entry in values)
 			{
 				Add(entry.Key, entry.Value);
 			}
 		}
 
-		public Arguments(object[] typedArguments, params IArgumentsComparer[] customComparers)
-			: this(customComparers)
+		public Arguments()
 		{
-			foreach (var @object in typedArguments)
-			{
-				if (@object == null)
-				{
-					throw new ArgumentNullException("typedArguments",
-					                                "Given array has null values. Only non-null values can be used as arguments.");
-				}
-
-				Add(@object.GetType(), @object);
-			}
+			dictionary = new Dictionary<object, object>(new ArgumentsComparer());
```

Comment:
Probably a micro-optimisation, but we could use a single instance of `ArgumentsComparer` for all `Arguments` rather than allocating a new one.

Reply:
Cool, made this a private static read-only field and passed the reference.